src/store/actions/movieDetailsAction.js

```diff
@@ -0,0 +1,16 @@
+import axios from "../axiosHelper";
+
+export const getMovieDetails = (id) => async (dispatch) => {
+  dispatch({
+    type: "LOADING_MOVIE_DETAILS",
+  });
+
+  const detailData = await axios.get(`/movies/${id}`);
+
+  dispatch({
+    type: "GET_MOVIE_DETAILS",
+    payload: {
+      movieDetail: detailData.data,
```

Comment:
If you're not planing to store anything else from this detailData in the payload, it is redundant to have "movieDetail" key in the payload, just put the resulting data as a payload.
This simplifies the logic "action results in data".